### DISCUSSION TITLE
Consume active tab value from verticals web part in custom code

### DISCUSSION THREAD

**Discussion by awdhaan** [microsoft-search/pnp-modern-search#4125] [Q&A]
Hi all, I'm currently working on a project to build an application that works with 2 distinct datasets. One is managed in a SharePoint lists where custom code determines user access to a list item and the other is document sets (and underlying files) that are configured with itemlevel permission.

To accomodate search, I want to use the PnP Search Verticals web part to distinquish between the 2 types of search. 
One tab for standard search through the document sets.
And another tab that will show a custom developed web part that can return the list data with the custom security trimming.
The custom web part should hide when the specific tab is not active.

Based on this, I am wondering if and how a custom web part could read the active tab from the PnP Search Verticals web part.
And if the same custom web part could read the searchterms from the PnP Search box web part.

I'm looking for some starter points to guide our developer. Help is much appreciated.

![Slide1](https://github.com/user-attachments/assets/0e1b0d15-f40f-40ea-a986-258ff3e123fd)
![Slide2](https://github.com/user-attachments/assets/a897c4b7-38e1-436f-8bd6-ede6825dcea7)

**Comment by patrikhellgren** [comment]
Hi @awdhaan, 
since you are not really clear about the custom security trimming it is a bit hard to recommend a specific way forward but since you seem to want to use PnP Modern Search (which I think could be a great fit for this) here is how I would set this up:

1. The Search box webpart and Documents vertical with a Search results webpart showing document sets is pretty self explanatory and straight forward. Set this up using out of the box Modern Search features.
2. Since you want custom security trimming for the list data you need to have that done server side or else the users could read the data on the client side using browser dev tools or other tools. I would use a [Custom data source](https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_data_sources/) Modern Search extension for this that is used by a second search results webpart connected to the List data vertical. A good option for this would be an Azure Function that requests the data from the original source using app permissions after authorizing the user and then security trimming the data before returning it to the Modern Search results webpart.
3. Now that you have got the correct data returned to the second Modern Search results webpart you want to display it in the List data vertical. Depending on how you would like it displayed you could:

- Use the out of the box layouts avaliable in the Search results webpart
- Modify one of the out of the box layouts using just handlebars
- Create a fully [custom layout](https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_layout/) using handlebars and [custom web components](https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_web_component/). Using this you can display the data in almost any way you want.

If you do it in this way you can use all available Modern Search features like search box, verticals, filters etc and you don't need to figure out the internals of SPFx dynamic data in the Modern Search webparts which you would if you were to develop your own custom webpart since that is the technology that is used and in quite a complex way too.

You can find basic examples of extending Modern Search here: [https://github.com/microsoft-search/pnp-modern-search-extensibility-samples](https://github.com/microsoft-search/pnp-modern-search-extensibility-samples)

**Reply by awdhaan** [reply]
Thanks for taking the time to help :-) 

The custom security trimming is basically that each list item contains a security string that includes references to security groups.

I will check with the developer if the custom source could be an option to integrate this. Thanks again.
